vms/platformvm/txs/builder/builder.go

```diff
@@ -125,6 +128,25 @@ type ProposalTxBuilder interface {
 		changeAddr ids.ShortID,
 	) (*txs.Tx, error)
 
+	// stakeAmount: amount the validator stakes
+	// startTime: unix time they start validating
+	// endTime: unix time they stop validating
+	// nodeID: ID of the node we want to validate with
+	// rewardAddress: address to send reward to, if applicable
+	// shares: 10,000 times percentage of reward taken from delegators
+	// keys: Keys providing the staked tokens
+	// changeAddr: Address to send change to, if there is any
+	NewAddPermissionlessValidatorTx(
```

Comment:
q: should these be some testing utilities already?
We plan to remove the whole txs/builder package from P-chain with the E-fork, so I am not sure it is worth adding stuff to it.
On the other hand such an utility is pretty helpful for tests as we currently build Permissionless stakers txs "manually" in UTs.

Reply:
I think the end goal is to use the P-chain wallet everywhere and remove this struct altogether. Just did the minimal thing here and added the missing funcs to unblock #2662

Reply:
I don't know that we'll actually use the P-chain wallet in the tests... It would either need to be moved into this package or there would be a circular dependency... I also think it makes sense for these to be testing utilities rather than added to this interface.

Reply:
Discussed offline- this struct will be moved to a testing utility post-Durango since we plan to remove the platformvm dependency on this struct.